pkg/dockerregistry/server/remoteblobgetter.go

```diff
@@ -236,7 +236,10 @@ func (rbgs *remoteBlobGetterService) findCandidateRepository(
 
 	// search the remaining registries for this digest
 	for _, repo := range repositoryCandidates {
-		spec := search[repo]
+		spec, ok := search[repo]
+		if !ok {
+			continue
```

Comment:
Did this crash for you? It would indicate an error in `identifyCandidateRepositories`.